### PR TITLE
ds9: update to 8.3

### PR DESCRIPTION
**Pull request by reneeotten**
#### Description
- update to latest upstream version

See: https://trac.macports.org/ticket/59843
Closes: https://trac.macports.org/ticket/61324
Closes: https://trac.macports.org/ticket/61642
Closes: https://trac.macports.org/ticket/64897

###### Tested on
macOS 10.15.7 19H1824 x86_64
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files? The package starts... nothing else tested

### CODE REVIEW

**Comment by lpsinger** [comment]
Thanks! Will try to review by Monday.

**Comment by reneeotten** [comment]
> Thanks! Will try to review by Monday.

Leo, it has been more than a week - did you have a chance to look at this. I'm inclined to merge this soonish as the current version doesn't build so this update cannot be much worse 😉

File: science/ds9/files/0001-Use-MacPorts-share-directory-instead-of-zipfs.patch
Comment (lpsinger) [inline science/ds9/files/0001-Use-MacPorts-share-directory-instead-of-zipfs.patch:65]: Would you please explain why the patches in ds9/unix/ds9.C are no longer needed?

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Comment by lpsinger** [comment]
The x11 variant does not start on my machine (macOS Monterey).

```
$ ds9
dyld[36313]: symbol not found in flat namespace '_astG3DAttr'
Abort trap: 6
```

File: science/ds9/files/0001-Use-MacPorts-share-directory-instead-of-zipfs.patch
Comment (reneeotten) [reply science/ds9/files/0001-Use-MacPorts-share-directory-instead-of-zipfs.patch:65]: hhmm... looks like I did mess that up indeed - they seems still to be required. Perhaps part of that patchfile (which touches two files) applied cleanly and the other part didn't, then when I looked at it the `ds9/unix/ds9.C` appeared correct. In any case, I agree that it should be put back - let me know if you want me to do that or if you take over from here.

**Comment by reneeotten** [comment]
> The x11 variant does not start on my machine (macOS Monterey).
> 
> ```
> $ ds9
> dyld[36313]: symbol not found in flat namespace '_astG3DAttr'
> Abort trap: 6
> ```

I am not sure... I did with the code in this PR:
```
% sudo port -dvt install ds9
```

```% port -v installed ds9
The following ports are currently installed:
  ds9 @8.3_0+x11 (active) requested_variants='' platform='darwin 19' archs='x86_64' date='2022-04-06T22:54:28-0400'
```

```
 % otool -L /opt/local/bin/ds9
/opt/local/bin/ds9:
	/opt/local/lib/libast.9.dylib (compatibility version 10.0.0, current version 10.1.0)
	/opt/local/lib/libast_err.9.dylib (compatibility version 10.0.0, current version 10.1.0)
	/opt/local/lib/libast_pal.9.dylib (compatibility version 10.0.0, current version 10.1.0)
	/opt/local/libexec/openssl10/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/local/libexec/openssl10/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/local/lib/libX11.6.dylib (compatibility version 11.0.0, current version 11.0.0)
	/opt/local/lib/libXss.1.dylib (compatibility version 2.0.0, current version 2.0.0, weak)
	/opt/local/lib/libXext.6.dylib (compatibility version 11.0.0, current version 11.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
	/opt/local/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.12)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1677.104.0)
	/opt/local/lib/libxml2.2.dylib (compatibility version 12.0.0, current version 12.13.0)
	/opt/local/lib/liblzma.5.dylib (compatibility version 8.0.0, current version 8.5.0)
	/opt/local/lib/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
	/opt/local/lib/libicui18n.67.dylib (compatibility version 67.0.0, current version 67.1.0)
	/opt/local/lib/libicuuc.67.dylib (compatibility version 67.0.0, current version 67.1.0)
	/opt/local/lib/libicudata.67.dylib (compatibility version 67.0.0, current version 67.1.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 902.1.0)
```

and the program starts just fine. I am still on Catalina, so there might be something specific with the latest macOS, I have no idea. However the first ticket about this port was submitted ~18 months ago, so it might be a good time to try and resolve this. I'm happy to help, but currently not able to reproduce the issue you're seeing. Perhaps comparing the output of the commands above on your systems might shed some light here?

**Comment by lpsinger** [comment]
The Aqua build crashes with the same error.

**Comment by lpsinger** [comment]
```
$ port installed ds9
The following ports are currently installed:
  ds9 @8.3_0+x11 (active)
$ otool -L /opt/local/bin/ds9
/opt/local/bin/ds9:
	/opt/local/lib/libast.9.dylib (compatibility version 10.0.0, current version 10.1.0)
	/opt/local/lib/libast_err.9.dylib (compatibility version 10.0.0, current version 10.1.0)
	/opt/local/lib/libast_pal.9.dylib (compatibility version 10.0.0, current version 10.1.0)
	/opt/local/libexec/openssl10/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/local/libexec/openssl10/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/local/lib/libX11.6.dylib (compatibility version 11.0.0, current version 11.0.0)
	/opt/local/lib/libXss.1.dylib (compatibility version 2.0.0, current version 2.0.0, weak)
	/opt/local/lib/libXext.6.dylib (compatibility version 11.0.0, current version 11.0.0)
	/opt/local/lib/libXft.2.dylib (compatibility version 6.0.0, current version 6.4.0)
	/opt/local/lib/libfontconfig.1.dylib (compatibility version 14.0.0, current version 14.0.0)
	/opt/local/lib/libfreetype.6.dylib (compatibility version 25.0.0, current version 25.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/opt/local/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.12)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1858.112.0)
	/opt/local/lib/libxml2.2.dylib (compatibility version 12.0.0, current version 12.13.0)
	/opt/local/lib/liblzma.5.dylib (compatibility version 8.0.0, current version 8.5.0)
	/opt/local/lib/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
	/opt/local/lib/libicui18n.67.dylib (compatibility version 67.0.0, current version 67.1.0)
	/opt/local/lib/libicuuc.67.dylib (compatibility version 67.0.0, current version 67.1.0)
	/opt/local/lib/libicudata.67.dylib (compatibility version 67.0.0, current version 67.1.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
```

**Comment by lpsinger** [comment]
Build log attached. [main.log](https://github.com/macports/macports-ports/files/8442996/main.log)

**Comment by reneeotten** [comment]
@lpsinger - I don't know why this doesn't work on Monterey but have no way of debugging this as I'm still on an older OS. I am not using this port myself and was just trying to help out the users who reported issues on Trac for a while now. I'll leave this to you as the maintainer to sort out with or without upstream to get it running on earlier OSses. 

Does the current MacPorts' version `ds9` port actually work for you on Monterey - if not, there isn't really a reason not to merge this as it fixes problems on other OSes?

**Comment by reneeotten** [comment]
@lpsinger I have corrected the part of the patchfile that inadvertently got removed... I'm inclined to merge this by the end of tomorrow unless you are committing to look into/fix the issues you're seeing on Monterey in the near future. 

In it's current state, `ds9` version 8.1 will not build on any recent system because of the `implicit function declarations`. At least that is fixed with this PR and it runs fine for me both with `x11` and `aqua` on Catalina. At the very least this will resolve some of the tickets that have been open for 1.5+ years with this port on Catalina.

Thoughts?

**Comment by lpsinger** [comment]
I'd rather not merge this hastily before getting it to build on Monterey, because it's a very complicated port and I want to make sure that we fully understand the changes.

**Comment by reneeotten** [comment]
> I'd rather not merge this hastily before getting it to build on Monterey, because it's a very complicated port and I want to make sure that we fully understand the changes.

This appears to be a simple update that works at least for Catalina, so I don't see a reason not to merge this. But you're the maintainer so I'll defer to you - but then please take care of fixing it as there have been open tickets for years, and recurring requests in these tickets, to fix the port.

**Comment by reneeotten** [comment]
> The patches are failing:
> 
> ```
> --->  Applying use-MacPorts-share-directory-instead-of-zipfs.patch
> Executing:  cd "/opt/local/var/macports/build/_Users_Shared_macports-staging_science_ds9/ds9/work/SAOImageDS9" && /usr/bin/patch -p0 < '/Users/Shared/macports-staging/science/ds9/files/use-MacPorts-share-directory-instead-of-zipfs.patch'
> patching file ds9/library/ds9.tcl
> can't find file to patch at input line 14
> Perhaps you used the wrong -p or --strip option?
> The text leading up to this was:
> --------------------------
> |--- unix/ds9.C.orig	2022-04-26 10:46:10.000000000 -0400
> |+++ unix/ds9.C	2022-04-26 10:48:59.000000000 -0400
> --------------------------
> File to patch: 
> Skip this patch? [y] 
> Skipping patch.
> 2 out of 2 hunks ignored
> Command failed:  cd "/opt/local/var/macports/build/_Users_Shared_macports-staging_science_ds9/ds9/work/SAOImageDS9" && /usr/bin/patch -p0 < '/Users/Shared/macports-staging/science/ds9/files/use-MacPorts-share-directory-instead-of-zipfs.patch'
> Exit code: 1
> Error: Failed to patch ds9: command execution failed
> ```

not anymore... there was a typo indeed. You can see the builds pass in the CI so that's all fixed.

**Comment by lpsinger** [comment]
Same error as before when I run ds9.

I tried updating the ast port, but I can't get it to build either :-(

**Comment by reneeotten** [comment]
okay, I tried to help out here... This update fixes stuff at least on some systems, but the maintainer appears to want everything fixed on the latest macOS versions as well before merging. That's fine with me, but I have personally no interest in this port and will not spend any time on this. So I'll close the PR and leave it to the maintainer to do the work to fix this port.

**Pull request closed without merging**
